<?xml version="1.0"?>
<launch>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find hearts_launcher)/rviz/rviz_mapping.rviz"/>

  <node name="hector_mapping" pkg="hector_mapping" type="hector_mapping">
    <param name="map_size"                   value="512" />
    <param name="map_resolution"             value="0.1" />
    <param name="pub_map_odom_transform"     value="true" />
    <param name="scan_topic"                 value="/hsrb/base_scan" />
    <param name="use_tf_scan_transformation" value="false" />
    <param name="map_update_angle_thresh"    value="2.0" />
    <param name="map_update_distance_thresh" value="0.10" />
    <param name="scan_subscriber_queue_size" value="1" />
    <param name="update_factor_free"         value="0.39" />
    <param name="update_factor_occupied"     value="0.85" />
    <param name="base_frame"                 value="base_range_sensor_link" />
  </node>


  <node name="mapping_pose_srv" pkg="hearts_robot_pose" type="mapping_pose_srv" args="" output="screen"/>
  <node name="top_pov_srv" pkg="hearts_top_pov" type="top_pov_srv" args="-conf /home/jesus/thesis_ws/src/mapping/hearts_tools/hearts_pov_conf.yaml" output="screen"/>
  <node name="top_map_srv" pkg="hearts_top_map" type="top_map_srv" args="" output="screen" />
  <node name="top_pov2map_srv" pkg="hearts_top_map" type="top_pov2map_srv" args="-debug" output="screen" />
  <node name="top_pov2map_cli" pkg="hearts_top_map" type="top_pov2map_cli" args="-debug" output="screen" />
  <node name="teleop" pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" >
  <remap from="/cmd_vel" to="/hsrb/command_velocity"/>

  </node>
</launch>
